### PR TITLE
Nodejs week3

### PR DESCRIPTION
**Pull request by aladdin91**
this is my node week 3 h/w

### CODE REVIEW

File: src/backend/api/meals.js [hunk 1]
```diff
@@ -1,7 +1,77 @@
...
+            query = query.limit(limit);
+        }
+        // Returns all meals sorted by the given key and direction
+        if (sortDir && sortDir === 'asc' || sortDir === 'desc') {
+
+            if (sortKey && (sortKey === 'price' || sortKey === 'when' || sortKey === 'max_reservations')) {
+                query = query.orderBy(sortKey, sortDir);
+            }
+        }
+        const sql = query.toString();
```
Comment (Divyajg) [inline src/backend/api/meals.js:58]: Why `query.toString()` ?

File: src/backend/api/reservations.js [hunk 2]
```diff
@@ -43,10 +81,25 @@ router.get('/:id', async(req, res) => {
...
 // Updates the reservation by id
+
+router.put('/:id', async (req, res) => {
+  try {
+    const id = req.params.id;
+    await knex('reservation')
+      .where({ id: id })
+      .update(req.body)
+
+    res.json({ "message": "reservation updated" });
```
Comment (Divyajg) [inline src/backend/api/reservations.js:97]: What happens if the `req.body` has some issue and failed to `Update`?
What is your code response?

**Review by Divyajg — COMMENTED** [review]
Well done. 👍
I left few comments.

File: src/backend/api/meals.js (hunk 1 above)
Comment (aladdin91) [reply src/backend/api/meals.js:58]: this was for debugging and forgot to remove it :)

File: src/backend/api/reservations.js (hunk 2 above)
Comment (aladdin91) [reply src/backend/api/reservations.js:97]: it will respond with console.error(err); and res.status(500).send(err), right?

**Comment by aladdin91** [comment]
thanks a lot for the review 🙏

**Pull request merged**